app/models/artefact.rb

```diff
@@ -344,7 +344,7 @@ def validate_redirect_url
 
   def valid_redirect_url_path?(target)
     URI.parse(target)
-    target.starts_with?("/") && target !~ %r{//} && target !~ %r{./\z}
+    target !~ %r{[^:]//}
```

Comment:
Whats this intended to be testing for?

I've found it seems to let through a few things that it shouldn't:

```
irb(main):001:0> "test" !~ %r{[^:]//}
=> true
irb(main):002:0> "/test" !~ %r{[^:]//}
=> true
irb(main):005:0> "//test" !~ %r{[^:]//}
=> true
```